refactor(walletconnect): migrate to WalletKit and implement multi-chain support

abretonc7s opened the pull request:
## **Description**
This PR completes the migration from WalletConnect SE SDK (a wrapper adding an unnecessary layer over WalletConnect) to WalletKit, improving our existing multi-chain support implementation.

### Motivation
1. WalletConnect SE SDK was an unnecessary abstraction layer over WalletConnect's core functionality
2. Our app already has initial multi-chain support capabilities that could be leveraged directly
3. Need for better session management and standardized error handling

### Solution
1. Removed WalletConnect SE SDK dependency in favor of WalletKit
2. Enhanced existing multi-chain support with:
   - Proper chain permission validation
   - Standardized CAIP-2 chain ID handling
   - Improved session management using namespaced permissions
3. Standardized error responses across WalletConnect interactions

Key implementation details:
- Added `getPermittedChains` utility for better chain permission management
- Updated session management to use proper namespacing
- Improved error handling with consistent response formats
- Updated test suites to cover new functionality

## **Related issues**

## **Manual testing steps**
1. Connect to a WalletConnect-enabled dApp
   - Verify connection establishes successfully
   - Check chain permissions are properly handled
2. Test multi-chain interactions
   - Switch between different networks
   - Verify proper permission handling
3. Test session management
   - Change accounts
   - Update sessions
   - Verify proper error handling
4. Test disconnection flow
   - Verify clean session termination
   - Check proper cleanup

## **Screenshots/Recordings**
### **Before**

### **After**

## **Pre-merge author checklist**
- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md)
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests with coverage for new functionality
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format
- [x] I've applied the right labels on the PR

## **Pre-merge reviewer checklist**
- [ ] I've manually tested the PR
- [ ] I confirm that this PR addresses all acceptance criteria and includes necessary testing evidence
- [ ] I've verified multi-chain support works as expected
- [ ] I've verified error handling and session management
- [ ] I've checked all dependency updates are compatible

abretonc7s committed:
wip

abretonc7s committed:
wip

abretonc7s committed:
wip

abretonc7s committed:
wip

abretonc7s committed:
wip

abretonc7s committed:
wip

abretonc7s committed:
unit tests

abretonc7s committed:
test

chakra-guy reviewed (dismissed):
Nicely done

abretonc7s committed:
wip

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

ignaciosantise commented on package.json:
```diff
@@ -242,8 +243,8 @@
+    "@walletconnect/jsonrpc-utils": "^1.0.8",
     "@walletconnect/react-native-compat": "2.15.0",
-    "@walletconnect/se-sdk": "1.11.0",
     "@walletconnect/utils": "^2.17.2",
```
"@reown/walletkit" already comes with:
```
    "@walletconnect/core": "2.18.0",
    "@walletconnect/jsonrpc-utils": "1.0.8",
    "@walletconnect/types": "2.18.0",
    "@walletconnect/utils": "2.18.0"
```

So it's safe to remove this from your dependencies
```
    "@walletconnect/core": "^2.17.2",
    "@walletconnect/jsonrpc-types": "^1.0.4",
    "@walletconnect/jsonrpc-utils": "^1.0.8",
```

chakra-guy committed:
pr comments

chakra-guy committed:
added tests

chakra-guy reviewed (commented):
_I have read the CLA Document and I hereby sign the CLA_

chakra-guy committed:
lint fix

chakra-guy reviewed (commented):
I have read the CLA Document and I hereby sign the CLA

chakra-guy committed:
lint fix

chakra-guy committed:
pr comment

chakra-guy committed:
Revert "pr comment"

This reverts commit 88a6ba29a8b69a23069db38124e1af04afd05b2c.

chakra-guy committed:
added back required deps

chakra-guy commented on package.json:
Hey @ignaciosantise, I added back those deps as they are needed. They are also present in the official docs:
> https://docs.reown.com/walletkit/web/installation
> https://docs.reown.com/walletkit/web/usage#initialization

ignaciosantise replied:
mmm okey. `@reown/walletkit:1.2.0` already comes with core and utils `2.18.0`. You'll see them in your yarn.lock file.

Now you also have 2.18.1, so you'll have two versions installed (check yarn.lock)

If you want to have them in your package.json because of typescript or something related, its ok, but remember to update their version whenever you update `@reown/walletkit` so you have just one version of each lib.
 
 this will unify versions
```
"@reown/walletkit": "1.2.1",
"@walletconnect/core": "2.18.1",
"@walletconnect/utils": "2.18.1",
```

chakra-guy replied:
Yeah, that's a good call! We will keep it sync for now!

ganchoradkov commented on package.json:
```diff
@@ -235,18 +235,17 @@
     "@react-navigation/native": "^5.9.4",
     "@react-navigation/stack": "^5.14.5",
     "@reduxjs/toolkit": "^1.9.7",
+    "@reown/walletkit": "^1.2.0",
```
```suggestion
    "@reown/walletkit": "^1.2.1",
```
new patch version is available for WalletKit that is using `2.18.1` `@walletconnect` deps. having all `@reown` and `@walletconnect` deps to latest will avoid having mismatched inner dependencies

ganchoradkov commented on package.json:
```diff
@@ -235,18 +235,17 @@
-    "@walletconnect/se-sdk": "1.11.0",
-    "@walletconnect/utils": "^2.17.2",
+    "@walletconnect/core": "^2.18.1",
+    "@walletconnect/react-native-compat": "2.18.0",
```
```suggestion
    "@walletconnect/react-native-compat": "2.18.1",
```
to keep all deps synced for consistency

chakra-guy committed:
pr comments

chakra-guy committed:
added more tests

christopherferreira9 commented:
Uploading Screen Recording 2025-02-20 at 10.59.11 AM.mov…

chakra-guy committed:
fix ui

chakra-guy reviewed (commented):
I have read the CLA Document and I hereby sign the CLA

chakra-guy committed:
Revert "fix ui"

This reverts commit bde66380644883716ce67d5006040189321c6ead.

christopherferreira9 commented:
Critical Issue ⚠️: When the connection starts from a non synced state, the wallet is not changing automatically to the proper chain to display a confirmation screen, this means users will be sending transactions for an unwanted chain. 

https://github.com/user-attachments/assets/4189f1d0-fae9-45ef-867d-6aea1c6990dc



**Note:**
Having a non synced state is normal, the wallet should change for the chain on the dapp in order to start processing a request. Here's a reference video:

https://github.com/user-attachments/assets/c0fd7f31-8b5b-4c4c-910e-a1b8e78a4776

christopherferreira9 commented:
SendTx seems to have the wrong origin in the pill

![Screenshot 2025-02-20 at 11 56 09 AM](https://github.com/user-attachments/assets/9c832a07-b3bd-46b1-9888-0514e16f31db)

christopherferreira9 commented:
Trying to sign after switching to a chain that hasn't been approved yet fails:

https://github.com/user-attachments/assets/eef268ec-026f-498f-94a8-24c24ceced17

Stacktrace:

![Screenshot 2025-02-20 at 11 59 16 AM](https://github.com/user-attachments/assets/9a4703f9-96ed-4585-81b2-b6258cceb749)

```{context: 'client'} 'request() -> isValidRequest() failed')```

``` {context: 'client'} 'Missing or invalid. request() chainId: eip155:11155111'```

```Uncaught (in promise) UnknownRpcError: An unknown RPC error occurred.```

chakra-guy commented:
> Critical Issue ⚠️: When the connection starts from a non synced state, the wallet is not changing automatically to the proper chain to display a confirmation screen, this means users will be sending transactions for an unwanted chain.
> 
>  Screen.Recording.2025-02-20.at.11.48.38.AM.mov 
> **Note:** Having a non synced state is normal, the wallet should change for the chain on the dapp in order to start processing a request. Here's a reference video:
> 
>  Screen.Recording.2025-02-20.at.11.52.04.AM.mov

Resolved! Regarding the critical issue where the wallet doesn’t automatically switch chains from a non-synced state:

**Root Cause**: This was a dapp-side bug, not a mobile app issue. When reconnecting after switching to Arbitrum, the dapp sent `personal_sign` requests with `chainId=eip155:1` (Ethereum) instead of `eip155:42161` (Arbitrum). The wallet correctly processes the chain specified in the request, but the dapp failed to update its state (even though from the UI it looks like its good, the actual request is with the wrong chain id).

**Verification**: Testing confirmed that when the dapp switches to Ethereum, then Arbitrum, and signs, the mobile app properly switches chains. 

Let me know if you need further clarification or additional test cases!

christopherferreira9 commented:
> > Critical Issue ⚠️: When the connection starts from a non synced state, the wallet is not changing automatically to the proper chain to display a confirmation screen, this means users will be sending transactions for an unwanted chain.
> > Screen.Recording.2025-02-20.at.11.48.38.AM.mov
> > **Note:** Having a non synced state is normal, the wallet should change for the chain on the dapp in order to start processing a request. Here's a reference video:
> > Screen.Recording.2025-02-20.at.11.52.04.AM.mov
> 
> Resolved! Regarding the critical issue where the wallet doesn’t automatically switch chains from a non-synced state:
> 
> **Root Cause**: This was a dapp-side bug, not a mobile app issue. When reconnecting after switching to Arbitrum, the dapp sent `personal_sign` requests with `chainId=eip155:1` (Ethereum) instead of `eip155:42161` (Arbitrum). The wallet correctly processes the chain specified in the request, but the dapp failed to update its state (even though from the UI it looks like its good, the actual request is with the wrong chain id).
> 
> **Verification**: Testing confirmed that when the dapp switches to Ethereum, then Arbitrum, and signs, the mobile app properly switches chains.
> 
> Let me know if you need further clarification or additional test cases!

The wallet should inform the dapp the current active chain, otherwise the dapp might think they're making a request for one chain when in reality it is requesting for another. In reality, regardless of what the dapp sends, the wallet needs to act as a safeguard and switch to the chain being requested. 
@ignaciosantise and @ganchoradkov , can you confirm what is the expected behavior in this situation?

ganchoradkov commented:
> > > Critical Issue ⚠️: When the connection starts from a non synced state, the wallet is not changing automatically to the proper chain to display a confirmation screen, this means users will be sending transactions for an unwanted chain.
> > > Screen.Recording.2025-02-20.at.11.48.38.AM.mov
> > > **Note:** Having a non synced state is normal, the wallet should change for the chain on the dapp in order to start processing a request. Here's a reference video:
> > > Screen.Recording.2025-02-20.at.11.52.04.AM.mov
> > 
> > 
> > Resolved! Regarding the critical issue where the wallet doesn’t automatically switch chains from a non-synced state:
> > **Root Cause**: This was a dapp-side bug, not a mobile app issue. When reconnecting after switching to Arbitrum, the dapp sent `personal_sign` requests with `chainId=eip155:1` (Ethereum) instead of `eip155:42161` (Arbitrum). The wallet correctly processes the chain specified in the request, but the dapp failed to update its state (even though from the UI it looks like its good, the actual request is with the wrong chain id).
> > **Verification**: Testing confirmed that when the dapp switches to Ethereum, then Arbitrum, and signs, the mobile app properly switches chains.
> > Let me know if you need further clarification or additional test cases!
> 
> The wallet should inform the dapp the current active chain, otherwise the dapp might think they're making a request for one chain when in reality it is requesting for another. In reality, regardless of what the dapp sends, the wallet needs to act as a safeguard and switch to the chain being requested. @ignaciosantise and @ganchoradkov , can you confirm what is the expected behavior in this situation?

The recommended approach is for the wallet to accept requests in any of the approved chains in the session. Users can switch the chain on the dapp multiple times before sending a request and the wallet will not get notified at all of these local (dapp side) chain switches because
- the chains are already approved by the user - asking them again to switch on the wallet is redundant 
- usually the wallet (on mobile device) is not active at the same time as the dapp

chakra-guy committed:
fix connection mismatch bug

christopherferreira9 commented:
> > > > Critical Issue ⚠️: When the connection starts from a non synced state, the wallet is not changing automatically to the proper chain to display a confirmation screen, this means users will be sending transactions for an unwanted chain.
> > > > Screen.Recording.2025-02-20.at.11.48.38.AM.mov
> > > > **Note:** Having a non synced state is normal, the wallet should change for the chain on the dapp in order to start processing a request. Here's a reference video:
> > > > Screen.Recording.2025-02-20.at.11.52.04.AM.mov
> > > 
> > > 
> > > Resolved! Regarding the critical issue where the wallet doesn’t automatically switch chains from a non-synced state:
> > > **Root Cause**: This was a dapp-side bug, not a mobile app issue. When reconnecting after switching to Arbitrum, the dapp sent `personal_sign` requests with `chainId=eip155:1` (Ethereum) instead of `eip155:42161` (Arbitrum). The wallet correctly processes the chain specified in the request, but the dapp failed to update its state (even though from the UI it looks like its good, the actual request is with the wrong chain id).
> > > **Verification**: Testing confirmed that when the dapp switches to Ethereum, then Arbitrum, and signs, the mobile app properly switches chains.
> > > Let me know if you need further clarification or additional test cases!
> > 
> > 
> > The wallet should inform the dapp the current active chain, otherwise the dapp might think they're making a request for one chain when in reality it is requesting for another. In reality, regardless of what the dapp sends, the wallet needs to act as a safeguard and switch to the chain being requested. @ignaciosantise and @ganchoradkov , can you confirm what is the expected behavior in this situation?
> 
> The recommended approach is for the wallet to accept requests in any of the approved chains in the session. Users can switch the chain on the dapp multiple times before sending a request and the wallet will not get notified at all of these local (dapp side) chain switches because
> 
> * the chains are already approved by the user - asking them again to switch on the wallet is redundant
> * usually the wallet (on mobile device) is not active at the same time as the dapp

Before checking @chakra-guy 's fix... Would there be a situation where wallet-kit component has a different chain being displayed than the one being triggered on the `Send TX` dapp button @ganchoradkov ? It could be a wallet-kit component bug. In the video previously attached you can see that:
- Wallet is on chainId 1
- Dropdown shows chainId 42161
- Send TX sends a tx for chainId 1

chakra-guy committed:
fix transaction confirm ui label

ganchoradkov commented:
>Before checking @chakra-guy 's fix... Would there be a situation where wallet-kit component has a different chain being displayed than the one being triggered on the Send TX dapp button @ganchoradkov ? It could be a wallet-kit component bug. In the video previously attached you can see that:
>Wallet is on chainId 1
Dropdown shows chainId 42161
Send TX sends a tx for chainId 1

this shouldn't be the case and is probably bug with the test app, will ping the team 🙏🙏🙏

chakra-guy committed:
fix bug

chakra-guy committed:
fix bug

Commit:
fix: walletconnect v2 connection

seems like normalizing the origin URL breaks the permission system, so we no longer normalize it

it seems like we request permissions with the non-normalized URL and then get permitted accounts
with the normalized URL, which results in no accounts being returned by engine

ecp4224 commented:
I have read the CLA Document and I hereby sign the CLA

chakra-guy committed:
fix: network add and switch

Commit:
fix(wc): track when we are handling request so we dont show loader

Commit:
chore(wc): remove un-needed changes

Commit:
fix(wc): always use session provided url if request is not verified

Commit:
chore(wc): remove unused imports from main merge

Commit:
chore(wc): bump version

metamaskbot committed:
Bump version number to 1652

Commit:
fix(wc): tests for WalletConnect2Session

Commit:
fix(wc): test for wc-utils

Commit:
Merge branch 'wcmigration' of github.com:MetaMask/metamask-mobile into wcmigration

Commit:
fix(wc): remove unused dep

Commit:
fix(wc): removed duplicated deps

Commit:
fix(wc): remove unused import

Commit:
fix(wc): typing in tests

Commit:
fix(wc): linting

Commit:
fix(wc): remove console logs

Commit:
fix(wc): add pageMeta with URL to not break chain switching

Commit:
chore(wc): bump version

metamaskbot committed:
Bump version number to 1655

Commit:
test(wc): add more tests for walletconnectv2 file

Commit:
Merge branch 'wcmigration' of github.com:MetaMask/metamask-mobile into wcmigration

Commit:
chore(wc): revert version bumps

Commit:
chore(wc): forgot to revert package json version

Commit:
chore(wc): add test coverage to ApprovalTagUrl + fix types

Commit:
chore(wc): fix lint errors

Commit:
chore(wc): push snapshot for approvaltagurl test

Commit:
fix(wc): fix snapshot?

Commit:
chore(wc): even more tests + fix diff

Commit:
chore(wc): ignore empty arrow func linting in tests

Commit:
chore(wc): add tests for removePendings

Commit:
chore(wc): added more tests to increase coverage

Commit:
chore(wc): added even more test coverage for wc-utils

Commit:
chore(wc): fix lint errors

Commit:
chore(wc): ignore any in test

Commit:
chore(wc): remove changes from podlock file

christopherferreira9 committed:
Merge branch 'main' into wcmigration

Commit:
chore(wc): re-add podfile.lock changes after running yarn setup

Commit:
Merge branch 'wcmigration' of github.com:MetaMask/metamask-mobile into wcmigration

Prithpal-Sooriya committed:
build: revert RTC-Folly pod checksum

christopherferreira9 committed:
Merge branch 'main' into wcmigration

Commit:
fix(wc): update approvaltagurl snapshot

Commit:
fix(wc): snapshot for approval tag url

ecp4224 committed:
Merge branch 'main' into wcmigration

Commit:
fix(wc): fix snapshot again

Commit:
Merge branch 'wcmigration' of github.com:MetaMask/metamask-mobile into wcmigration

ecp4224 committed:
Merge branch 'main' into wcmigration

tommasini commented on app/core/Multichain/constants.ts:
```diff
@@ -9,6 +9,8 @@ import BTC from '../../images/bitcoin-logo.png';
 import SOL from '../../images/solana-logo.png';
 import { MultichainBlockExplorerFormatUrls } from './networks';
 
+export const EVM_IDENTIFIER = 'eip155';
```
https://github.com/MetaMask/utils/blob/48267f8e18de9a5480d473c0aa99624bc0468410/src/caip-types.ts#L116
Can we re use the identifier defined at metamask utils instead? Just to keep one source of truth

tommasini commented on app/core/WalletConnect/WalletConnect2Session.ts:
```diff
@@ -218,23 +227,99 @@ class WalletConnect2Session {
+    this.isHandlingChainChange = true;
+
+    try {
+      // const origin = normalizeOrigin(this.session.peer.metadata.url);
```
I see a lot of commented code on this handleChainChange function, any reason for it?

tommasini commented on app/core/WalletConnect/WalletConnect2Session.ts:
```diff
@@ -218,23 +227,99 @@ class WalletConnect2Session {
+  isHandlingRequest = () => this._isHandlingRequest;
+
+  emitEvent = async (eventName: string, data: unknown) => {
+    //console.log('🔵 emitEvent', eventName, data);
```
can we remove console logs commented?

tommasini commented on app/core/WalletConnect/WalletConnect2Session.ts:
```diff
@@ -218,23 +227,99 @@ class WalletConnect2Session {
-      await this.handleSwitchEthereumChain(initialRequest);
+      const chainIdHex = initialRequest.params.request.params[0].chainId;
+      const chainIdDecimal = parseInt(chainIdHex, 16);
+      //console.log('👉👉👉 approveRequest chainIdDecimal', chainIdDecimal);
```
Here as well

tommasini commented on app/core/WalletConnect/wc-utils.ts:
```diff
@@ -101,14 +147,209 @@ export const waitForNetworkModalOnboarding = async ({
 };
-export default parseWalletConnectUri;
+
+// export const normalizeOrigin = (origin: string): string => {
```
Is this commented for any reason?

Commit:
chore(wc): remove un-needed comments

Commit:
Merge branch 'wcmigration' of github.com:MetaMask/metamask-mobile into wcmigration

Commit:
chore(wc): replace EVM_IDENTIFIER with KnownCaipNamespace enum

chakra-guy commented:
I have read the CLA Document and I hereby sign the CLA

ffmcgee725 reviewed (dismissed):
Thorough review seems to have already taken place, so reviewed code owned by `wallet-api-platform-team`, and LGTM 👍🏾

Commit:
wc(fix): permissions getPermittedChains

Commit:
fix(wc): approvaltag snapshot

ecp4224 committed:
Merge branch 'main' into wcmigration

Commit:
fix(wc): more and more and more test

Commit:
Merge branch 'wcmigration' of github.com:MetaMask/metamask-mobile into wcmigration

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      expect(pairedSessions.length).toBeGreaterThan(0);
+    });
+
+    it('should not connect if URI is invalid', async () => {
```
```suggestion
    it('does not connect if URI is invalid', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      });
+
+      const sessions = manager.getSessions();
+      expect(sessions.length).toBe(1);
```
Q: why 1? Where does the existing session come from in this test?

ecp4224 replied:
the session comes from mock iirc

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      );
+    });
+
+    it('should handle existing sessionTopic connections', async () => {
```
```suggestion
    it('shows loading indicator for existing sessionTopic connections', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      expect(showLoadingSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should handle WalletConnect v1 URIs', async () => {
```
```suggestion
    it('creates new session for WalletConnect v1 URIs', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      );
+    });
+
+    it('should handle invalid URIs', async () => {
```
```suggestion
    it('logs a warning to console on invalid URIs', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      mockWeb3Wallet = (manager as unknown as { web3Wallet: IWalletKit }).web3Wallet;
+    });
+
+    it('should handle session requests through event emission', async () => {
```
C: not sure how to rename this test. I don't understand what it does very well. Please find a great name matching our guidleines that helps understanding it.

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      expect(session).toBeDefined();
+    });
+
+    it('should reject invalid session requests through event emission', async () => {
```
S: what do you mean by "through event emission"?

```suggestion
    it('rejects invalid session requests through event emission', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -265,6 +269,222 @@ describe('WC2Manager', () => {
+      });
+    });
+
+    it('should handle errors during session request processing', async () => {
```
```suggestion
    it('logs an error to console on session request error', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
-    it('should handle v2 URIs correctly', async () => {
-      const mockWcUri = 'wc://testv2uri';
+  describe('WC2Manager isWalletConnect', () => {
+    it('should return true for valid WalletConnect origins', () => {
```
```suggestion
    it('returns true for valid WalletConnect origins', () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
-        wcUri: mockWcUri,
-        redirectUrl: 'https://example.com',
-        origin: AppConstants.DEEPLINKS.ORIGIN_DEEPLINK,
+    it('should return false for invalid WalletConnect origins', () => {
```
```suggestion
    it('returns false for invalid WalletConnect origins', () => {
```

Commit:
fix(wc): wow even more tests

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
+  });
+
+  describe('WC2Manager session proposal handling', () => {
+    it('should handle session proposal rejection', async () => {
```
S: rename to comply to guidelines.
I suggest somehting like this, but please clarify it with the real wording, not sure it's an event, wallet, the right terms...

```suggestion
    it('returns rejectSession event to wallet on proposal rejection', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
 
-      const pairedSessions = manager.getSessions();
-      expect(pairedSessions.length).toBeGreaterThan(0);
+    it('should handle session proposal with invalid wallet status', async () => {
```
```suggestion
    it('logs "invalid wallet status" error to console on session proposal with invalid wallet status', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
+      consoleSpy.mockRestore();
+    });
+
+    it('should remove all pending session proposals', async () => {
```
```suggestion
    it('removes all pending session proposals', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
 
-      const sessions = manager.getSessions();
-      expect(sessions.length).toBe(1);
+    it('should handle errors when removing pending session proposals', async () => {
```
```suggestion
    it('logs errors to console when removing pending session proposals fails', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
 
-    it('should handle deeplink sessions correctly', async () => {
-      const mockWcUri = 'wc://testv2deeplinkuri';
+    it('should remove all pending session requests', async () => {
```
```suggestion
    it('removes all pending session requests', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
 
-      const pairedSessions = manager.getSessions();
-      expect(pairedSessions.length).toBeGreaterThan(0);
+    it('should handle errors when removing pending session requests', async () => {
```
```suggestion
    it('logs error to console when removing pending session requests fails', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
+      );
+    });
+
+    it('should handle empty pending proposals and requests', async () => {
```
```suggestion
    it('does not process empty pending proposals and requests', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -304,51 +524,453 @@ describe('WC2Manager', () => {
+      jest.clearAllMocks();
+    });
+
+    it('should handle session delete event for a deeplink session', async () => {
```
S: rename this test and all the following...

NicolasMassart commented on app/core/WalletConnect/wc-utils.test.ts:
C: this test file is exactly what I am expecting. Make the others look like this 👍

ecp4224 committed:
Update app/components/UI/ApprovalTagUrl/ApprovalTagUrl.test.tsx

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/Permissions/index.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Merge branch 'main' into wcmigration

Commit:
Merge branch 'main' into wcmigration

Commit:
fix(wc): update test names to use present tense verbs

Commit:
wc(test): update test snapshot

Commit:
wc(fix): package issues

Commit:
chore(wc): update podlock file

Commit:
fix(wc): removed duplicated packages again

Commit:
chore(wc): revert rct-folly in podfile.lock

ecp4224 committed:
Merge branch 'main' into wcmigration

NicolasMassart commented on app/core/WalletConnect/WalletConnect2Session.test.ts:
```diff
@@ -169,19 +169,19 @@ describe('WalletConnect2Session', () => {
   });
 
-  it('should handle request correctly and reject invalid chainId', async () => {
+  it('handles request correctly and rejects invalid chainId', async () => {
```
S: focus on the result and avoid weasel words like "handle" that caries no meaning and all meanings at the same time. Favour precise language. Here the request working is part of the implementation. The fact that it rejects invalid chainid is the expectation, use this:
```suggestion
  it('rejects invalid chainId', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnect2Session.test.ts:
```diff
@@ -265,7 +265,7 @@ describe('WalletConnect2Session', () => {
   });
 
-  it('should handle session update correctly', async () => {
+  it('handles session update correctly', async () => {
```
```suggestion
  it('updates session', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -838,7 +838,7 @@ describe('WC2Manager', () => {
     });
 
-    it('should handle session delete event for a deeplink session', async () => {
+    it('handles session delete event for a deeplink session', async () => {
```
S: avoid "handle"
```suggestion
    it('processes delete event for deeplink session', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -861,7 +861,7 @@ describe('WC2Manager', () => {
     });
 
-    it('should handle session delete event for a non-deeplink session', async () => {
+    it('handles session delete event for a non-deeplink session', async () => {
```
S: avoid "handle"
```suggestion
    it('processes delete event for non-deeplink session', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -880,7 +880,7 @@ describe('WC2Manager', () => {
     });
 
-    it('should handle session delete event for non-existent session', async () => {
+    it('handles session delete event for non-existent session', async () => {
```
```suggestion
    it('processes delete event for non-existent session', async () => {
```

NicolasMassart commented on app/core/WalletConnect/WalletConnectV2.test.ts:
```diff
@@ -898,7 +898,7 @@ describe('WC2Manager', () => {
     });
 
-    it('should handle errors during storage update', async () => {
+    it('handles errors during storage update', async () => {
```
```suggestion
    it('deletes session on storage update error', async () => {
```

NicolasMassart reviewed (commented):
Some suggestion to update unit test names to avoid "handle" and clarify expectations

christopherferreira9 commented:
@SocketSecurity ignore npm/ox@0.6.7

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnect2Session.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

ecp4224 committed:
Update app/core/WalletConnect/WalletConnectV2.test.ts

Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>

NicolasMassart reviewed (approved):
Nice work, I approve this.

christopherferreira9 reviewed (approved):
Approving from SDK

ganchoradkov reviewed (approved):
Awesome work 🔥🔥

adonesky1 commented on app/core/RPCMethods/lib/ethereum-chain-utils.js:
```diff
@@ -241,6 +241,9 @@ export async function switchToNetwork({
       networkConfiguration.shortName,
     ticker: networkConfiguration.ticker || 'ETH',
     chainColor: networkConfiguration.color,
+    pageMeta: {
+      url: origin,
+    },
```
Should we add a test in `ethereum-chain-utils.test.ts` and `wallet_switchEthereumChain.test.js` to ensure that this property is being properly added?

ecp4224 replied:
Good catch, and I think we should add one. For the sake of getting this PR finally merged, I will follow up on this PR to add tests 👍

jiexi reviewed (approved):
Approving for `ethereum-chain-utils.js`
